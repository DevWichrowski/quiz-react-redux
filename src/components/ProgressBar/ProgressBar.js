import React from "react";
import "./ProgressBar.scss";
import {quizQuestions} from "../../core/questions";
import PropTypes from "prop-types";


const ProgressBar = ({step}) => {
    const questionsAmount = quizQuestions.length;

    return (
        <div className="progress-container">
            <div className="progress-bar" style={{width: step * questionsAmount + '%'}}>
                <div className="progress-bar__step">
                    {step}/{questionsAmount}
                </div>
            </div>
        </div>
    )
};

ProgressBar.propTypes = {
    step: PropTypes.number
};

export default ProgressBar;

