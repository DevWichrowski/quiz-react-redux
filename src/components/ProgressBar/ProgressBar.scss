@import "../../styles/colors";
@import "../../styles/mixins";
@import "../../styles/media-queries";

.progress-container {
  height: 10px;
  width: 80%;
  @include boxShadow();
  border-radius: 10px;
  margin: 30px auto;
  background-color: $light-gray;

  .progress-bar {
    transition: .5s;
    background: $gradient-blue;
    color: $gray;
    font-size: 8px;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    border-radius: 8px;
    height: 100%;

    &__step {
      margin-right: 10px;
      font-weight: bold;
    }
  }
}