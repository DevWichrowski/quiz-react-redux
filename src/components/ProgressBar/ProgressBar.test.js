import {testComponentRender} from "../../core/helpers/tests/test-component-render";
import ProgressBar from "./ProgressBar";
import React from 'react';

describe('ProgressBar component', () => {
    testComponentRender(<ProgressBar/>);
});