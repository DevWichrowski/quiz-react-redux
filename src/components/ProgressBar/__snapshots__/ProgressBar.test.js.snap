// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should render 1`] = `
<div
  className="progress-container"
>
  <div
    className="progress-bar"
    style={
      Object {
        "width": "NaN%",
      }
    }
  >
    <div
      className="progress-bar__step"
    >
      /
      10
    </div>
  </div>
</div>
`;
